scripts/flowcell_summary_uppload_LIMS.py

```diff
@@ -42,68 +42,75 @@ def get_run_qcs(fc, lanesobj):
     fc_db = couch['flowcells']
+    xfc_db = couch['x_flowcells']
+    process_dict = {'hiseq':'Illumina Sequencing (Illumina SBS) 4.0', 'miseq':'MiSeq Run (MiSeq) 4.0', 'hiseqx':'Illumina Sequencing (HiSeq X) 1.0'}    
+    # Collect flowcell processes based upon how script is called
     if all_flowcells:
-        flowcells = lims.get_processes(type = ['Illumina Sequencing (Illumina SBS) 4.0','MiSeq Run (MiSeq) 4.0'])
-        for fc in flowcells:
-            try:
-                closed = date(*map(int, fc.date_run.split('-')))
-                delta = today-closed
-            except AttributeError:
-                #Happens if fc has no date run, we should just not update and get to the next flowcell
+        flowcells = lims.get_processes(type = process_dict.values())
+    elif flowcell:
+        if not run_type:
+            raise SystemExit("Option -f should always be called with option -t. kindly refer -h fo rmore info.")        
+        fc_id = flowcell if run_type == 'miseq' else flowcell[1:]
+        try:
+            flowcells = [lims.get_processes(type = process_dict[run_type], udf = {'Flow Cell ID' : fc_id})[0]]
+            days = float('inf') #no need of days check when a flowcell is specified
+        except:
+            raise SystemExit("Could not find any process for FC %s (type: %s)" % (flowcell, run_type))
+    
+    for fc in flowcells:
+        try:
+            closed = date(*map(int, fc.date_run.split('-')))
+            delta = today-closed
+            if not delta.days < days:
                 continue
+        except AttributeError:
+            #Happens if fc has no date run, we should just not update and get to the next flowcell
+            continue
+        
+        fc_is_hiseqx = False
+        fc_udfs = dict(fc.udf.items())
+        try:
+            flowcell_name = fc_udfs['Flow Cell ID']
+            if not '-' in flowcell_name:
+                flowcell_name = "%s%s" % (fc_udfs['Flow Cell Position'],fc_udfs['Flow Cell ID'])
+                if 'HiSeq X' in fc_udfs['SBS Kit Type']:
+                    fc_is_hiseqx = True
+        except KeyError:
+            continue
 
-            #if delta.days < days and dict(fc.udf.items()).has_key('Flow Cell ID'):
-            if dict(fc.udf.items()).has_key('Flow Cell ID'):
-                if '-' in dict(fc.udf.items())['Flow Cell ID']:
-                    flowcell_name = dict(fc.udf.items())['Flow Cell ID']
-                elif dict(fc.udf.items()).has_key('Flow Cell Position'):
-                    flowcell_name = dict(fc.udf.items())['Flow Cell Position'] + dict(fc.udf.items())['Flow Cell ID'] 
-                key = find_flowcell_from_view(fc_db, flowcell_name)
-                if key:
-                    dbobj = fc_db.get(key)
-                    print dbobj['modification_time']+'  '+key
-                    if delta.days < days:
-                        dbobj["illumina"]["run_summary"] = get_sequencing_info(fc) #located in genologics.lims_utils
```

Comment:
changed this key to `dbobj["lims_data"]["run_summary"]` (*line 95*), as @vezzi thought would be more appropriate place/key for it(which is). But did not a `if` condition, so for time being some documents in `flowcell` database might have dual entry (after deployed) i.e. `dbobj["lims_data"]["run_summary"] ` and `dbobj["illumina"]["run_summary"] ` which will eventually go away.

But if it is not preferred I can add an `if` and check for time being.